### PR TITLE
Add file generation for service containers

### LINKED ISSUE #1: Command to create new service from template
As a user, I want to be able to quickly create a new service from a template. I would also like the ability to not automatically include it in the repository so I can have my own set of services locally.

### PR DESCRIPTION
**Pull request by joshminnie**
Closes #1 

Adds file generation for service containers. Users are now able to execute the following command to create new service containers:
```
$ workspace service new mysql80 -i mysql:8.0 --ignore
```
The name of the service starts from a base image, identified by the `-i` argument above, and can be included in the repo or ignored.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,7 +1,14 @@
...
+workspace.json
 
 # Keep data folders in the repository, but not the contents
 **/data/*
 !**/data/.gitkeep
+
+# Ignore all services unless specifically allowlisted
+services/*
+# BEGIN allowlisting services
+!services/mysql57
```
Comment (joshpennington) [inline .gitignore]: This is a nice touch to only track the services you want to track

File: lib/generators/services/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+version: "3.9"
+
+services:
+  <%= config[:service] %>:
+    container_name: <%= config[:service] %>
+    build: "./"
+    platform: "linux/x86_64"
```
Comment (joshpennington) [inline lib/generators/services/docker-compose.yml]: You may want to make this an optional thing to generate. Most of the services out there (that I have encountered) have binaries available for both x86_64 as well as arm64. The big omission is MySQL.

**Review by joshpennington — DISMISSED** [review]
I left a few notes in here and left a few issues that I ran into while getting started that I don't think would stop this going to main, but this works as expected.

File: lib/generators/services/docker-compose.yml (hunk 2 above)
Comment (joshminnie) [reply lib/generators/services/docker-compose.yml]: @joshpennington I've converted that to an optional param now.

**Review by joshpennington — APPROVED** [review]
Works as expected!

**Pull request merged**